gluecodium/src/main/resources/templates/cpp/CppStruct.mustache

```diff
@@ -51,10 +51,17 @@ public:
 }}{{#if attributes.cpp.accessors}}
 {{#set structElement=this}}{{#fields}}
 {{prefixPartial "cpp/CppFieldDoc" "    "}}
-    {{resolveName typeRef}} {{resolveName this "" "getter"}}( ) const { return {{resolveName}}; }{{!!
+    {{#ifPredicate typeRef "needsRefSuffix"}}const {{resolveName typeRef}}&{{!!
+}} {{resolveName this "" "getter"}}( ) const & { return {{resolveName}}; }
+    {{resolveName typeRef}} {{resolveName this "" "getter"}}( ) const && { return {{resolveName}}; }{{/ifPredicate}}{{!!
```

Comment:
```suggestion
    const {{resolveName typeRef}}&&{{resolveName this "" "getter"}}( ) const && { return std::move({{resolveName}}); }{{/ifPredicate}}{{!!
```

Reply:
Fixed the problem by introducing 3 overloads of getter for non-trivial types.

Moreover, for such types 2 overload of setters have been added -- `const lvalue` ref and `rvalue ref`.

Reply:
The code was updated according to the offline discussion.